addons/intuit-quickbooks/template/index.js

```diff
@@ -0,0 +1,61 @@
+const qbo = require('./quickbooks.js');
+const health = require('./health.js');
+const util = require('util');
+const Superagent = require('superagent');
+const Sdk = require('@fusebit/add-on-sdk');
+const QuickBooks = require('node-quickbooks');
+
+let client;
+
+module.exports = (ctx, callback) => {
+  if (!client) {
+    client = new QuickBooks(
+      null, // Client ID, behavior is overridden
+      null, // Client Secret, behavior is overridden
+      null, // Access token, will be obtained
+      false, // No token secret for Auth0
+      ctx.configuration.intuit_company_id, // Company ID
+      ctx.configuration.intuit_environment.toLowerCase() === 'sandbox', // Use production account
+      false, // Turn debugging off
+      null, // Use latest minor version
+      '2.0', // OAuth version
+      ctx.configuration.intuit_refresh_token
+    );
+
+    QuickBooks.setOauthVersion('2.0', client.useSandbox);
+
+    // Use Fusebit Connector to get access token
+    client.refreshAccessToken = (cb) => {
+      Superagent.post(ctx.configuration.intuit_get_token_url)
+        .send({
+          refresh_token: ctx.configuration.intuit_refresh_token,
+        })
+        .then((res) => {
+          client.refreshToken = res.body.refresh_token;
+          client.token = res.body.access_token;
+          if (cb) cb(null, res.body);
+        })
+        .catch((err) => {
+          if (cb) cb(err);
+        });
+    };
+
+    client.refreshAccessToken((err) => {
+      invoke(ctx, callback);
+    });
+  } else {
+    invoke(ctx, callback);
```

Comment:
Invert this if block to reduce the overall indentation of the rest of the function.

Reply:
Fixed